### PR TITLE
Allow psternary to draw lines and polygons

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

If **-S** is not given the we can plot lines or polygons.  Polygons are expected if a fill (**-G**) or a CPT (**-C**) is given, else we assume lines.  Polygons are plotted before the gridlines while symbols and lines are plotted after (similar to **psxy**).  Added a simple test example to make sure it works (_ternary_pol.sh_).

### CODE REVIEW

File: doc/rst/source/explain_symbols_ternary.rst_ [hunk 1]
```diff
@@ -1,5 +1,6 @@
 **-S**\ [*symbol*][*size*]
-    Plot individual symbols in a ternary diagram.
+    Plot individual symbols in a ternary diagram.  if **-S** is not given then
```
Comment (seisman) [inline doc/rst/source/explain_symbols_ternary.rst_:2]:
```suggestion
    Plot individual symbols in a ternary diagram.  If **-S** is not given then
```

File: src/psternary.c [hunk 2]
```diff
@@ -518,6 +518,24 @@ EXTERN_MSC int GMT_psternary (void *V_API, int mode, void *args) {
...
+		if (GMT_Open_VirtualFile (API, GMT_IS_DATASET, GMT_IS_POLYGON, GMT_IN|GMT_IS_REFERENCE, D, vfile) == GMT_NOTSET) {
+			GMT_Report (API, GMT_MSG_ERROR, "Unable to create a virtual data set\n");
+			Return (API->error);
+		}
+		sprintf (cmd, "-R0/1/0/1 -JX%gi -O -K %s", width, vfile);
+		if (Ctrl->C.active) {strcat (cmd, " -C"); if (Ctrl->C.string) strcat (cmd, Ctrl->C.string);}
+		else if (Ctrl->G.active) {strcat (cmd, " -G"); strcat (cmd, Ctrl->G.string);}
+		if (Ctrl->W.active) {strcat (cmd, " -W"); strcat (cmd, Ctrl->W.string);}
+		if ((error = GMT_Call_Module (API, "psxy", GMT_MODULE_CMD, cmd))) {
+			GMT_Report (API, GMT_MSG_ERROR, "Unable to plot symbols\n");
```
Comment (seisman) [inline src/psternary.c:532]:
```suggestion
			GMT_Report (API, GMT_MSG_ERROR, "Unable to plot polygons\n");
```

File: src/psternary.c [hunk 3]
```diff
@@ -563,6 +584,20 @@ EXTERN_MSC int GMT_psternary (void *V_API, int mode, void *args) {
...
 	}
+	else if (Ctrl->W.active && !Ctrl->S.active && !(Ctrl->G.active || Ctrl->C.active)) {	/* Plot lines */
+		char vfile[GMT_VF_LEN] = {""};
+		if (GMT_Open_VirtualFile (API, GMT_IS_DATASET, GMT_IS_LINE, GMT_IN|GMT_IS_REFERENCE, D, vfile) == GMT_NOTSET) {
+			GMT_Report (API, GMT_MSG_ERROR, "Unable to create a virtual data set\n");
+			Return (API->error);
+		}
+		sprintf (cmd, "-R0/1/0/1 -JX%gi -O -K -W%s %s", width, vfile, Ctrl->W.string);
+		if ((error = GMT_Call_Module (API, "psxy", GMT_MODULE_CMD, cmd))) {
+			GMT_Report (API, GMT_MSG_ERROR, "Unable to plot symbols\n");
```
Comment (seisman) [inline src/psternary.c:595]:
```suggestion
			GMT_Report (API, GMT_MSG_ERROR, "Unable to plot lines\n");
```

File: test/psternary/ternary_pol.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+# Test ternary diagram plotting with (a,b,c-coordinate polygons
```
Comment (seisman) [inline test/psternary/ternary_pol.sh:2]:
```suggestion
# Test ternary diagram plotting with a,b,c-coordinate polygons
```

**Review by seisman — APPROVED** [review]
Looks good, except a few typos.

**Comment by PaulWessel** [comment]
Sorry for being a bit sloppy and thanks for finding those typos.

**Pull request merged**